pipeline/config_template.txt
<OBJECT>Exoplanet
<OBJECT-NAME>Kepler-10b
<OBJECT-DATE>2020/06/09 03:45
<OBJECT-DIAMETER>12252.726496466576
<OBJECT-GRAVITY>5.518951671719958
<OBJECT-GRAVITY-UNIT>rho
<OBJECT-OBS-LONGITUDE>179.56
<OBJECT-STAR-DISTANCE>0.3153298478037273
<OBJECT-STAR-VELOCITY>0.00235
<OBJECT-SOLAR-LONGITUDE>0.0
<OBJECT-SOLAR-LATITUDE>0.00
<OBJECT-STAR-TYPE>K
<OBJECT-STAR-TEMPERATURE>4250.888264596154
<OBJECT-STAR-RADIUS>0.6030311875219362
<OBJECT-OBS-LATITUDE>5.20
<OBJECT-OBS-VELOCITY>-98.740
<OBJECT-PERIOD>0.83749070
<OBJECT-STAR-METALLICITY>-0.15
<OBJECT-INCLINATION>84.80
<OBJECT-SEASON>180
<OBJECT-ORBIT>0.00000000,0.00000,0.83749070,0.00000,0.01685,2455034.296243
<OBJECT-POSITION-ANGLE>0.00
<GEOMETRY-USER-PARAM>45
<GEOMETRY-OBS-ALTITUDE>13.878009908590041
<GEOMETRY>Observatory
<GEOMETRY-OFFSET-NS>0.0
<GEOMETRY-OFFSET-EW>0.0
<GEOMETRY-OFFSET-UNIT>arcsec
<GEOMETRY-ALTITUDE-UNIT>pc
<GEOMETRY-STELLAR-TYPE>G
<GEOMETRY-STELLAR-TEMPERATURE>5777
<GEOMETRY-STELLAR-MAGNITUDE>0
<GEOMETRY-SOLAR-ANGLE>89.997
<GEOMETRY-PLANET-FRACTION>1.000e+00
<GEOMETRY-STAR-DISTANCE>0.000000e+00
<GEOMETRY-STAR-FRACTION>1.612878e-04
<GEOMETRY-REF>User
<GEOMETRY-DISK-ANGLES>1
<GEOMETRY-ROTATION>-1.52,0.57
<GEOMETRY-OBS-ANGLE>48.121
<GEOMETRY-AZIMUTH>0.000
<GEOMETRY-BRDFSCALER>1.000
<ATMOSPHERE-NGAS>5
<ATMOSPHERE-GAS>CO2,H2O,N2O,CH4,H2S
<ATMOSPHERE-TYPE>HIT[2],HIT[1],HIT[4],HIT[6],HIT[31]
<ATMOSPHERE-ABUN>1,1,1,1,1
<ATMOSPHERE-UNIT>scl,scl,scl,scl,scl
<ATMOSPHERE-DESCRIPTION>Venus atmosphere - Ehrenreich et al. 2012, A&amp;amp;amp;A
<ATMOSPHERE-STRUCTURE>Equilibrium
<ATMOSPHERE-WEIGHT>37.64315194048416
<ATMOSPHERE-PRESSURE>2.445881112454545
<ATMOSPHERE-PUNIT>mbar
<ATMOSPHERE-LAYERS-MOLECULES>CO2,H2O,N2O,CH4,H2S
<ATMOSPHERE-LAYERS>50
<ATMOSPHERE-LAYER-1>2.445881112454545,293.4922148015887,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-2>1.8115866510469205,290.91511912453456,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-3>1.3417848388215097,287.2919106721205,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-4>0.9938175204872575,282.8176408859863,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-5>0.736089151890339,277.81177016127003,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-6>0.5451979144672221,272.6217670475991,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-7>0.4038108226103715,267.5622853742694,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-8>0.2990898830136086,262.8964114051871,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-9>0.2215264007607025,258.8325743151707,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-10>0.16407758677600778,255.51448416888024,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-11>0.12152707031664133,253.00610113025223,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-12>0.09001125083529968,251.28680928359867,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-13>0.06666848181088576,250.2652869947045,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-14>0.049379232326203094,249.80728355199372,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-15>0.03657363298059998,249.76580469724598,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-16>0.027088931244680696,250.0043780906165,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-17>0.020063913157554746,250.4100183170623,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-18>0.014860704822858116,250.89722572557756,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-19>0.011006853254294747,251.40622769539058,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-20>0.008152427492889153,251.89849460089007,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-21>0.006038244763632339,252.35159678724122,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-22>0.004472336596349425,252.75450957640146,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-23>0.003312518027012571,253.1038032507887,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-24>0.0024534771573856626,253.40077657724362,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-25>0.0018172128008740285,253.6494187302064,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-26>0.0013459519497541212,253.8550291061644,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-27>0.0009969039675351173,254.02332740308424,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-28>0.0007383751854357128,254.15991242207693,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-29>0.0005468911070895274,254.26995943274963,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-30>0.0004050649167429929,254.35807473251634,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-31>0.0003000187508061784,254.42824954427587,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-32>0.00022221438370682308,254.48387339149497,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-33>0.0001645871539479334,254.52778028725592,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-34>0.00012190449057708302,254.56231045546602,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-35>9.029079406500497e-05,254.58937680223593,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-36>6.687553062480652e-05,254.6105297528411,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-37>4.9532586823077015e-05,254.62701697219228,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-38>3.6687217797948665e-05,254.6398363485975,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-39>2.71730599203644e-05,254.64978177198182,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-40>2.0126224602319077e-05,254.65748191358887,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-41>1.4906856935880987e-05,254.66343257879888,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-42>1.104103666224703e-05,254.66802337106154,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-43>8.17774606018103e-06,254.67155944954422,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-44>6.056997424297681e-06,254.67427913677832,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-45>4.486226097993622e-06,254.67636806843456,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-46>3.3228055408415086e-06,254.67797049632887,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-47>2.461096792955875e-06,254.6791982706549,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-48>1.8228564235401304e-06,254.68013794564524,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-49>1.350132002265012e-06,254.68085637829043,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-LAYER-50>1e-06,254.68140512407612,1.0,1.0,1.0,1.0,1.0
<ATMOSPHERE-GCM-PARAMETERS>72,46,40,180.0,-90.0,5.00,4.00,Winds,Tsurf,Psurf,Albedo,Temperature,Pressure,N2,H2O,CO2,N2O,CH4,Water,WaterIce
<ATMOSPHERE-TEMPERATURE>210
<ATMOSPHERE-TAU>0.07,0.07,0.07,0.07,0.07,0.07
<ATMOSPHERE-NAERO>0
<ATMOSPHERE-NMAX>0
<ATMOSPHERE-LMAX>0
<ATMOSPHERE-CONTINUUM>Rayleigh,Refraction,CIA_all,UV_all
<SURFACE-NSURF>0
<SURFACE-TEMPERATURE>293.4922148015887
<SURFACE-ALBEDO>0.18273453588425762
<SURFACE-EMISSIVITY>0.8172654641157424
<SURFACE-GAS-RATIO>1.0
<SURFACE-GAS-UNIT>ratio
<SURFACE-MODEL>Lambert
<SURFACE-PHASEMODEL>ISO
<GENERATOR-INSTRUMENT>user
<GENERATOR-RANGE1>0.4
<GENERATOR-RANGE2>20
<GENERATOR-RANGEUNIT>um
<GENERATOR-RESOLUTION>200
<GENERATOR-RESOLUTIONUNIT>RP
<GENERATOR-TELESCOPE>SINGLE
<GENERATOR-DIAMTELE>5.64
<GENERATOR-BEAM>1.0
<GENERATOR-BEAM-UNIT>diffrac
<GENERATOR-TELESCOPE1>1
<GENERATOR-TELESCOPE2>2.0
<GENERATOR-TELESCOPE3>1.0
<GENERATOR-NOISE>NO
<GENERATOR-NOISE1>1
<GENERATOR-NOISE2>1
<GENERATOR-NOISEOTEMP>300
<GENERATOR-NOISEOEFF>1
<GENERATOR-NOISEOEMIS>1
<GENERATOR-NOISETIME>360
<GENERATOR-NOISEFRAMES>100
<GENERATOR-NOISEPIXELS>1
<GENERATOR-TRANS-APPLY>N
<GENERATOR-TRANS-SHOW>N
<GENERATOR-LOGRAD>N
<GENERATOR-GAS-MODEL>Y
<GENERATOR-CONT-MODEL>Y
<GENERATOR-CONT-STELLAR>N
<GENERATOR-RADUNITS>ppm
<GENERATOR-TRANS>02-01
<GENERATOR-GCM-BINNING>13
<GENERATOR-RESOLUTIONKERNEL>N
<GEOMETRY-TEMISSION>756.000
<OBJECT-SHAPE>Sphere


archive/legacy_physics/temp_config.txt
<OBJECT>Exoplanet
<OBJECT-NAME>Kepler-10b
<OBJECT-DATE>2020/06/09 03:45
<OBJECT-DIAMETER>16313.135033883429
<OBJECT-GRAVITY>6.48670315832474
<OBJECT-GRAVITY-UNIT>rho
<OBJECT-OBS-LONGITUDE>179.56
<OBJECT-STAR-DISTANCE>1.4635702184922526
<OBJECT-STAR-VELOCITY>0.00235
<OBJECT-SOLAR-LONGITUDE>0.0
<OBJECT-SOLAR-LATITUDE>0.00
<OBJECT-STAR-TYPE>G
<OBJECT-STAR-TEMPERATURE>5489.392416556529
<OBJECT-STAR-RADIUS>0.902184646172457
<OBJECT-OBS-LATITUDE>5.20
<OBJECT-OBS-VELOCITY>-98.740
<OBJECT-PERIOD>0.83749070
<OBJECT-STAR-METALLICITY>-0.15
<OBJECT-INCLINATION>84.80
<OBJECT-SEASON>180
<OBJECT-ORBIT>0.00000000,0.00000,0.83749070,0.00000,0.01685,2455034.296243
<OBJECT-POSITION-ANGLE>0.00
<GEOMETRY-USER-PARAM>45
<GEOMETRY-OBS-ALTITUDE>3.644047868914301
<GEOMETRY>Observatory
<GEOMETRY-OFFSET-NS>0.0
<GEOMETRY-OFFSET-EW>0.0
<GEOMETRY-OFFSET-UNIT>arcsec
<GEOMETRY-ALTITUDE-UNIT>pc
<GEOMETRY-STELLAR-TYPE>G
<GEOMETRY-STELLAR-TEMPERATURE>5777
<GEOMETRY-STELLAR-MAGNITUDE>0
<GEOMETRY-SOLAR-ANGLE>89.997
<GEOMETRY-PLANET-FRACTION>1.000e+00
<GEOMETRY-STAR-DISTANCE>0.000000e+00
<GEOMETRY-STAR-FRACTION>1.612878e-04
<GEOMETRY-REF>User
<GEOMETRY-DISK-ANGLES>1
<GEOMETRY-ROTATION>-1.52,0.57
<GEOMETRY-OBS-ANGLE>48.121
<GEOMETRY-AZIMUTH>0.000
<GEOMETRY-BRDFSCALER>1.000
<ATMOSPHERE-NGAS>7
<ATMOSPHERE-GAS>O2,N2,H2,CO2,H2O,CH4,NH3
<ATMOSPHERE-TYPE>HIT[7],HIT[22],HIT[45],HIT[2],HIT[1],HIT[6],HIT[11]
<ATMOSPHERE-ABUN>1,1,1,1,1,1,1
<ATMOSPHERE-UNIT>scl,scl,scl,scl,scl,scl,scl
<ATMOSPHERE-DESCRIPTION>Venus atmosphere - Ehrenreich et al. 2012, A&amp;amp;amp;A
<ATMOSPHERE-STRUCTURE>Equilibrium
<ATMOSPHERE-WEIGHT>13.657204034610993
<ATMOSPHERE-PRESSURE>1.0255530027823907
<ATMOSPHERE-PUNIT>mbar
<ATMOSPHERE-LAYERS-MOLECULES>O2,N2,H2,CO2,H2O,CH4,NH3
<ATMOSPHERE-LAYERS>50
<ATMOSPHERE-LAYER-1>1.0255530027823907,194.87211866261092,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-2>0.7731886960480061,193.00921622872056,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-3>0.5829252686838127,190.67780470127144,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-4>0.43948116495613826,188.0422316234106,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-5>0.33133525809801256,185.29815196174263,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-6>0.24980149779532265,182.64109737501173,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-7>0.1883312650123331,180.2711538413258,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-8>0.1419874007729419,178.42314766670333,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-9>0.10704766400276551,177.36829412657497,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-10>0.08070576900533505,177.34467783444163,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-11>0.060845990535339754,178.4508067148244,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-12>0.04587323322551972,180.58823192685517,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-13>0.0345849169032872,183.49843322205876,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-14>0.026074387896902584,186.85798422942207,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-15>0.01965809853177702,190.36678966546668,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-16>0.014820706028192503,193.79514574884897,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-17>0.0111736812601196,196.99279969849786,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-18>0.008424102918258503,199.87727593533862,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-19>0.006351130690535904,202.415839806991,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-20>0.004788267835717023,204.60860990945707,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-21>0.0036099885175922197,206.47539535724744,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-22>0.002721655835527459,208.0464070263201,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-23>0.0020519207889340475,209.35613038790294,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-24>0.001546991676573884,210.43951278066757,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-25>0.0011663136609830402,211.32974355923517,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-26>0.0008793114897736779,212.0570853974421,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-27>0.0006629337560843748,212.6483751267363,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-28>0.0004998014583765463,213.12693511581801,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-29>0.0003768121558189727,213.5127238047651,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-30>0.00028408760797566464,213.82261431708764,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-31>0.0002141803754444363,214.07073070777986,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-32>0.00016147565728896282,214.26879840581609,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-33>0.0001217403221130636,214.42648315607846,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-34>9.178291190771032e-05,214.5517042994108,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-35>6.919731089946352e-05,214.65091561580016,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-36>5.2169491424849634e-05,214.72935158340215,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-37>3.9331815068388074e-05,214.79123970613722,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-38>2.9653186840096582e-05,214.83998115923634,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-39>2.2356239808531e-05,214.87830281505416,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-40>1.6854898634393032e-05,214.9083840232338,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-41>1.2707307239890073e-05,214.93196152007388,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-42>9.580339863893677e-06,214.95041565918285,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-43>7.222845105971021e-06,214.96484087388322,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-44>5.445473977542872e-06,214.97610295664626,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-45>4.105471791937325e-06,214.9848854072619,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-46>3.0952124101414523e-06,214.99172678087086,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-47>2.3335539371402734e-06,214.9970506714787,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-48>1.759321576671373e-06,215.00118970200643,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-49>1.3263942010847956e-06,215.0044046601476,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-LAYER-50>1e-06,215.00689971962814,0.040885258466005325,0.06607979536056519,0.3177936375141144,5.715466855349405e-08,0.2154405266046524,0.1535840481519699,0.20621664822101593
<ATMOSPHERE-GCM-PARAMETERS>72,46,40,180.0,-90.0,5.00,4.00,Winds,Tsurf,Psurf,Albedo,Temperature,Pressure,N2,H2O,CO2,N2O,CH4,Water,WaterIce
<ATMOSPHERE-TEMPERATURE>210
<ATMOSPHERE-TAU>0.07,0.07,0.07,0.07,0.07,0.07
<ATMOSPHERE-NAERO>0
<ATMOSPHERE-NMAX>0
<ATMOSPHERE-LMAX>0
<ATMOSPHERE-CONTINUUM>Rayleigh,Refraction,CIA_all,UV_all
<SURFACE-NSURF>0
<SURFACE-TEMPERATURE>194.87211866261092
<SURFACE-ALBEDO>0.5904519350366879
<SURFACE-EMISSIVITY>0.4095480649633121
<SURFACE-GAS-RATIO>1.0
<SURFACE-GAS-UNIT>ratio
<SURFACE-MODEL>Lambert
<SURFACE-PHASEMODEL>ISO
<GENERATOR-INSTRUMENT>user
<GENERATOR-RANGE1>0.4
<GENERATOR-RANGE2>20
<GENERATOR-RANGEUNIT>um
<GENERATOR-RESOLUTION>200
<GENERATOR-RESOLUTIONUNIT>RP
<GENERATOR-TELESCOPE>SINGLE
<GENERATOR-DIAMTELE>5.64
<GENERATOR-BEAM>1.0
<GENERATOR-BEAM-UNIT>diffrac
<GENERATOR-TELESCOPE1>1
<GENERATOR-TELESCOPE2>2.0
<GENERATOR-TELESCOPE3>1.0
<GENERATOR-NOISE>NO
<GENERATOR-NOISE1>1
<GENERATOR-NOISE2>1
<GENERATOR-NOISEOTEMP>300
<GENERATOR-NOISEOEFF>1
<GENERATOR-NOISEOEMIS>1
<GENERATOR-NOISETIME>360
<GENERATOR-NOISEFRAMES>100
<GENERATOR-NOISEPIXELS>1
<GENERATOR-TRANS-APPLY>N
<GENERATOR-TRANS-SHOW>N
<GENERATOR-LOGRAD>N
<GENERATOR-GAS-MODEL>Y
<GENERATOR-CONT-MODEL>Y
<GENERATOR-CONT-STELLAR>N
<GENERATOR-RADUNITS>ppm
<GENERATOR-TRANS>02-01
<GENERATOR-GCM-BINNING>13
<GENERATOR-RESOLUTIONKERNEL>N
<GEOMETRY-TEMISSION>756.000
<OBJECT-SHAPE>Sphere


pypsg/settings.py
"""
PyPSG settings
==============

This module allows users to configure PyPSG.
"""
from pathlib import Path
import warnings
import json
from astropy import units as u

REQUEST_TIMEOUT = 120

PSG_URL = 'https://psg.gsfc.nasa.gov/api.php'
PSG_PORT = 3000
INTERNAL_PSG_URL = f'http://localhost:{PSG_PORT}/api.php'

USER_DATA_PATH = Path.home() / '.pypsg'
USER_SETTINGS_PATH = USER_DATA_PATH / 'settings.json'

DEFAULT_SETTINGS = {
    'url': PSG_URL,
    'api_key': "API_KEY",
    'encoding': 'utf-8',
    'cfg_max_lines': 1500,
    'timeout': REQUEST_TIMEOUT
}


settings_need_reload = False
def save_settings(**kwargs):
    if not USER_DATA_PATH.exists():
        USER_DATA_PATH.mkdir()
    if not USER_SETTINGS_PATH.exists():
        USER_SETTINGS_PATH.touch()
    with USER_SETTINGS_PATH.open('r') as file:
        try:
            previous_settings = json.load(file)
        except json.decoder.JSONDecodeError:
            previous_settings = {}
        
    for key, value in kwargs.items():
        if key in DEFAULT_SETTINGS.keys():
            previous_settings[key] = value
        else:
            raise KeyError(f'Unknown setting {key}.')
    with USER_SETTINGS_PATH.open('w') as file:
        json.dump(previous_settings, file, indent=4)
    # print(f'Saved settings to {USER_SETTINGS_PATH}')
    # pylint: disable-next=global-statement
    global settings_need_reload
    settings_need_reload = True
    # print('Reloading settings...')
    reload_settings()

def load_settings():
    try:
        with USER_SETTINGS_PATH.open('r') as file:
            try:
                settings = json.load(file)
            except json.decoder.JSONDecodeError:
                settings = {}
    except FileNotFoundError:
        settings = {}
    for key, value in DEFAULT_SETTINGS.items():
        if key not in settings:
            settings[key] = value
    # pylint: disable-next=global-statement
    global settings_need_reload
    settings_need_reload = False
    return settings

user_settings = load_settings()

def reload_settings():
    # pylint: disable-next=global-statement
    global user_settings
    user_settings = load_settings()

class StaleSettingsWarning(RuntimeWarning):
    pass

def get_setting(key):
    if settings_need_reload:
        msg = 'Your user settings have changed recently.\n'
        msg += 'Please reload the settings using the `pypsg.settings.reload_settings()` function.'
        warnings.warn(msg,StaleSettingsWarning) 
    if key in user_settings:
        return user_settings[key]
    else:
        raise KeyError(f'Unknown setting {key}.')
    



################################
# Some settings related to PSG #
################################

psg_pressure_unit = u.bar
"""
PSG atmospheric pressure unit.

This unit is determined by PSG and used to standardize
the atmospheric pressure of planets in VSPEC.

:type: astropy.units.Unit
"""
psg_aerosol_size_unit = u.m
"""
PSG aerosol size unit.

This unit is determined by PSG and used to
standardize aerosol size in VSPEC.

:type: astropy.units.Unit
"""

atmosphere_type_dict = {'H2':45,'He':0,'H2O':1,'CO2':2,'O3':3,'N2O':4,'CO':5,'CH4':6,'O2':7,
                        'NO':8,'SO2':9,'NO2':10,'N2':22,'HNO3':12,'HO2NO2':'SEC[26404-66-0] Peroxynitric acid',
                        'N2O5':'XSEC[10102-03-1] Dinitrogen pentoxide','O':'KZ[08] Oxygen',
                        'OH':'EXO[OH]'}
"""
A dictionary mapping molecular species to the default
database to use to create opacities. These are all
internal to PSG, but must be set by ``VSPEC``.

Integers mean that we want to use data from the HITRAN database, which for a number ``N``
is represented in PSG by ``HIT[N]``. Strings are sent straight to PSG as is.

:type: dict
"""

aerosol_name_dict = {
    'Water':{
        'name':'CLDLIQ',
        'size':'REL'
    },
    'WaterIce':{
        'name':'CLDICE',
        'size':'REI'
    }
}
"""
A dictionary mapping aerosol species from their PSG name
to their name in the WACCM NetCDF format.

:type: dict
"""

aerosol_type_dict = {
    'Water': 'AFCRL_Water_HRI',
    'WaterIce': 'Warren_ice_HRI'
}
"""
A dictionary mapping aerosol species to the default
database to use. These are all
internal to PSG, but must be set by ``VSPEC``.

:type: dict
"""
